[CI] StIntersectionTests testCrankyEvaluateBlockWithoutNulls {TestCase=geo_shape and geo_shape} failing

**Build Scans:**
- [elasticsearch-intake #42082 / part3](https://gradle-enterprise.elastic.co/s/czfl4jpilxegu)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.function.scalar.spatial.StIntersectionTests.testCrankyEvaluateBlockWithoutNulls {TestCase=geo_shape and geo_shape}" -Dtests.seed=D5BA76CEF15585A0 -Dtests.locale=fr-SN -Dtests.timezone=Antarctica/Vostok -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.expression.function.scalar.spatial.StIntersectionTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCrankyEvaluateBlockWithoutNulls%20%7BTestCase%3Dgeo_shape%20and%20geo_shape%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Unexpected value at position '0'
Expected: <[1 7 0 0 0 0 0 0 0]>
     but: was <[1 2 0 0 0 0 0 0 0]>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testCrankyEvaluateBlockWithoutNulls {TestCase=geo_shape and geo_shape}
- [main] 1 consecutive failures in step part3
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -704,9 +704,6 @@ tests:
 - class: org.elasticsearch.xpack.search.AsyncSearchActionIT
   method: testSearchPhaseFailureLeak
   issue: https://github.com/elastic/elasticsearch/issues/152117
-- class: org.elasticsearch.xpack.esql.expression.function.scalar.spatial.StIntersectionTests
-  method: testCrankyEvaluateBlockWithoutNulls {TestCase=geo_shape and geo_shape}
-  issue: https://github.com/elastic/elasticsearch/issues/152119
 - class: org.elasticsearch.index.engine.InternalEngineTests
   method: testLookupSeqNoByIdInLucene
   issue: https://github.com/elastic/elasticsearch/issues/152120

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/SpatialBinaryGeometryBlockProcessor.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/SpatialBinaryGeometryBlockProcessor.java
@@ -129,7 +129,7 @@ class SpatialBinaryGeometryBlockProcessor {
      * For anything else (a true heterogeneous collection), pre-flatten to a supported type
      * using a self-union so the binary operation can proceed.
      */
-    private static Geometry flattenIfHeterogeneousCollection(Geometry geom) {
+    static Geometry flattenIfHeterogeneousCollection(Geometry geom) {
         if (geom instanceof MultiPoint || geom instanceof MultiLineString || geom instanceof MultiPolygon) {
             return geom;
         }

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/AbstractBinarySpatialGeometryFunctionTestCase.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/AbstractBinarySpatialGeometryFunctionTestCase.java
@@ -31,6 +31,7 @@ import static org.elasticsearch.xpack.esql.core.type.DataType.CARTESIAN_SHAPE;
 import static org.elasticsearch.xpack.esql.core.type.DataType.GEO_POINT;
 import static org.elasticsearch.xpack.esql.core.type.DataType.GEO_SHAPE;
 import static org.elasticsearch.xpack.esql.core.util.SpatialCoordinateTypes.UNSPECIFIED;
+import static org.elasticsearch.xpack.esql.expression.function.scalar.spatial.SpatialBinaryGeometryBlockProcessor.flattenIfHeterogeneousCollection;
 
 /**
  * Base test case for spatial functions that take two geometry arguments and return a geometry
@@ -137,8 +138,8 @@ public abstract class AbstractBinarySpatialGeometryFunctionTestCase extends Abst
             return null;
         }
         try {
-            Geometry leftGeom = UNSPECIFIED.wkbToJtsGeometry(leftWkb);
-            Geometry rightGeom = UNSPECIFIED.wkbToJtsGeometry(rightWkb);
+            Geometry leftGeom = flattenIfHeterogeneousCollection(UNSPECIFIED.wkbToJtsGeometry(leftWkb));
+            Geometry rightGeom = flattenIfHeterogeneousCollection(UNSPECIFIED.wkbToJtsGeometry(rightWkb));
             Geometry result = jtsOp.apply(leftGeom, rightGeom);
             return UNSPECIFIED.jtsGeometryToWkb(result);
         } catch (Exception e) {

